src/renderer/lib/hooks/useDolphin.ts

```diff
@@ -0,0 +1,103 @@
+/* eslint-disable react-hooks/rules-of-hooks */
+import {
+  ipc_clearDolphinCache,
+  ipc_configureDolphin,
+  ipc_importDolphinSettings,
+  ipc_launchNetplayDolphin,
+  ipc_reinstallDolphin,
+  ipc_viewSlpReplay,
+} from "@dolphin/ipc";
+import { DolphinLaunchType, ReplayQueueItem } from "@dolphin/types";
+import electronLog from "electron-log";
+import { MainEndpointResponse } from "ipc";
+import { useToasts } from "react-toast-notifications";
+import create from "zustand";
+import { combine } from "zustand/middleware";
+
+const log = electronLog.scope("useDolphin");
+
+export const useDolphinStore = create(
+  combine(
+    {
+      netplayDolphinOpen: false,
+      playbackDolphinOpen: false,
+    },
+    (set) => ({
+      setDolphinOpen: (dolphinType: DolphinLaunchType, val = true) => {
+        if (dolphinType === DolphinLaunchType.NETPLAY) {
+          set({ netplayDolphinOpen: val });
+        } else {
+          set({ playbackDolphinOpen: val });
+        }
+      },
+    }),
+  ),
+);
+
+export const useDolphin = () => {
+  const { addToast } = useToasts();
+  const setDolphinOpen = useDolphinStore((store) => store.setDolphinOpen);
+
+  const handleResult = (res: MainEndpointResponse<any>, errLog: string): boolean => {
```

Comment:
```suggestion
  const handleResult = <T>(res: MainEndpointResponse<T>, errLog: string): boolean => {
```

Should probably just use a generic for this.